habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py

```diff
@@ -825,3 +825,128 @@ def get_object_regions(
     )
 
     return sim.semantic_scene.get_regions_for_points(key_points)
+
+
+def get_link_normalized_joint_position(
+    objectA: habitat_sim.physics.ManagedArticulatedObject, link_ix: int
+) -> float:
+    """
+    Normalize the joint limit range [min, max] -> [0,1] and return the current joint state in this range.
+
+    :param objectA: The parent ArticulatedObject of the link.
+    :param link_ix: The index of the link within the parent object. Not the link's object_id.
+
+    :return: normalized joint position [0,1]
+    """
+
+    assert objectA.get_link_joint_type(link_ix) in [
+        habitat_sim.physics.JointType.Revolute,
+        habitat_sim.physics.JointType.Prismatic,
+    ], f"Invalid joint type '{objectA.get_link_joint_type(link_ix)}'. Open/closed not a valid check for multi-dimensional or fixed joints."
+
+    joint_pos_ix = objectA.get_link_joint_pos_offset(link_ix)
+    joint_pos = objectA.joint_positions[joint_pos_ix]
+    limits = objectA.joint_position_limits
+
+    # compute the normalized position [0,1]
+    n_pos = (joint_pos - limits[0][joint_pos_ix]) / (
+        limits[1][joint_pos_ix] - limits[0][joint_pos_ix]
+    )
+    return n_pos
+
+
+def set_link_normalized_joint_position(
+    objectA: habitat_sim.physics.ManagedArticulatedObject,
+    link_ix: int,
+    normalized_pos: float,
+) -> None:
+    """
+    Set the joint's state within its limits from a normalized range [0,1] -> [min, max]
+
+    Assumes the joint has valid joint limits.
+
+    :param objectA: The parent ArticulatedObject of the link.
+    :param link_ix: The index of the link within the parent object. Not the link's object_id.
+    :param normalized_pos: The normalized position [0,1] to set.
+    """
+
+    assert objectA.get_link_joint_type(link_ix) in [
+        habitat_sim.physics.JointType.Revolute,
+        habitat_sim.physics.JointType.Prismatic,
+    ], f"Invalid joint type '{objectA.get_link_joint_type(link_ix)}'. Open/closed not a valid check for multi-dimensional or fixed joints."
+
+    joint_pos_ix = objectA.get_link_joint_pos_offset(link_ix)
+    limits = objectA.joint_position_limits
+    joint_positions = objectA.joint_positions
+    joint_positions[joint_pos_ix] = limits[0][joint_pos_ix] + (
+        normalized_pos * (limits[1][joint_pos_ix] - limits[0][joint_pos_ix])
+    )
+    objectA.joint_positions = joint_positions
+
+
+def link_is_open(
```

Comment:
I personally don't think providing `link_is_open` and `link_is_closed` is necessary.

Reply:
@jacobkrantz don't we need these queries for state evaluation? 
Agreed that these are very light wrappers, so they aren't strictly necessary. Really just convenience/clarity to call an obvious API.
I think these will mostly be necessary if we decide to handle axis inversion programmatically rather than enforcing that assets conform to our assumptions.

Reply:
This will allow us to check if a particular drawer is open in a chest of drawers for example, correct? We should keep it for now and remove later if we dont end up using it.